src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -313,7 +313,7 @@ internal partial class UnboundLambda
             ImmutableArray<string> names,
             bool isAsync,
             bool hasErrors = false)
-            : base(BoundKind.UnboundLambda, syntax, null, hasErrors || !types.IsDefault && types.Any(SymbolKind.ErrorType))
+            : base(BoundKind.UnboundLambda, syntax, null, hasErrors || !types.IsDefault && types.Any(t => t.Type?.Kind == SymbolKind.ErrorType))
```

Comment:
>t => t.Type?.Kind == SymbolKind.ErrorType [](start = 101, length = 41)

It is not obvious why this change is needed. Could you please elaborate?

Reply:
I removed two `SymbolExtensions.Any` methods. One was unused and the other was only used in this one location, so I replaced this use with the equivalent.

---
In reply to: [285212378](https://github.com/dotnet/roslyn/pull/35765#discussion_r285212378) [](ancestors = 285212378)